samcli/lib/build/app_builder.py

```diff
@@ -430,6 +430,11 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             LOG.debug("%s image is built for %s function", build_image, function_name)
         except docker.errors.BuildError as ex:
             LOG.error("Failed building function %s", function_name)
+            for log in ex.build_log:
+                if "stream" in log:
+                    self._stream_writer.write_str(log["stream"])
+                elif "error" in log:
+                    self._stream_writer.write_str(log["error"])
```

Comment:
Seems we have a very similar function here 
https://github.com/aws/aws-sam-cli/blob/a18782ac09006914664a064bcd5c8964c2c97ca7/samcli/lib/build/app_builder.py#L443

I wonder if it's possible to reuse this function in this place
```suggestion
            self._stream_lambda_image_build_logs(ex.build_log, function_name)
```

Reply:
Looked at that initially, but rejected because the `_stream_lambda_image_build_logs` uses `LogStreamer._stream_write` which will throw a `LogStreamError` if it sees an error rather than logging it. Provided the logs are always of the form `<several 'stream' errors>,<one 'error' log>` this will work, otherwise not.

This is incompatible with @jonife's suggestion from above BTW. Which one should I build?

Reply:
Thanks for the clarification. Yes you are right. This function won't work on logs that has multiple error. Maybe we can consider adding a param in `LogStreamer ` to suppress the error. In this way it will also address @jonife concerns

Reply:
Adding the param would work, I'll have a look at that. It won't address @jonife's concerns though, as `LogStreamer` doesn't write anything about what sort of log it is. My experiments with it so far indicate that for the usual Docker log patterns it literally does what my patch did (minus the error problems).